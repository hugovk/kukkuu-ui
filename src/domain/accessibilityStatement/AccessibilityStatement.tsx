import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './accessibilityStatement.module.scss';
import { getCurrentLanguage } from '../../common/translation/TranslationUtils';
import AccessibilityStatementEn from './AccessibilityStatementEn';
import AccessibilityStatementFi from './AccessibilityStatementFi';
import AccessibilityStatementSv from './AccessibilityStatementSv';
import PageWrapper from '../app/layout/PageWrapper';

type StatementProps = {
  lang: string;
};

const Statement: FunctionComponent<StatementProps> = props => {
  const lang = props.lang;
  switch (lang) {
    case 'en':
      return <AccessibilityStatementEn />;
    case 'fi':
      return <AccessibilityStatementFi />;
    case 'sv':
      return <AccessibilityStatementSv />;
    default:
      return <p>Invalid language.</p>;
  }
};

const AccessibilityStatement: FunctionComponent = () => {
  const { i18n } = useTranslation();
  const locale = getCurrentLanguage(i18n);

  return (
    <PageWrapper title={'accessibilityStatement.title'}>
      <div className={styles.accessibilityStatement}>
        <Statement lang={locale} />
      </div>
    </PageWrapper>
  );
};

export default AccessibilityStatement;
