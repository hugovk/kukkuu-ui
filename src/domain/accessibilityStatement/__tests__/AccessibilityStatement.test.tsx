import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AccessibilityStatement from '../AccessibilityStatement';
import AccessibilityStatementFi from '../AccessibilityStatementFi';

it('renders snapshot correctly', () => {
  const element = shallow(
    <AccessibilityStatement>
      <AccessibilityStatementFi />
    </AccessibilityStatement>
  );
  expect(toJson(element)).toMatchSnapshot();
});
