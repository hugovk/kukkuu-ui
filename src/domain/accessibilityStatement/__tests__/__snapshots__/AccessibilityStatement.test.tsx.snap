// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<PageWrapper
  title="accessibilityStatement.title"
>
  <div
    className="accessibilityStatement"
  >
    <Statement
      lang="fi"
    />
  </div>
</PageWrapper>
`;
